variables.tf

```diff
@@ -27,7 +28,30 @@ variable "additional_queries" {
     description = string
     time_window = number
   }))
-  description = "List of additional alert rule queries to create with a file path, description and time_window"
-  default = {}
-  nullable = false
+  description = "List of additional alert rule queries to create with a file path, description and time_window."
+  default     = {}
+  nullable    = false
 }
+
+variable "event_pipeline_key" {
+  type        = string
+  description = "Function key provided by monitoring team."
+  sensitive   = true
+}
+
+variable "automation_account" {
+  type = object({
+    name                = string
+    id                  = string
+    location            = string
+    resource_group_name = string
+  })
+  description = "Automation account where the resource graph script will be deployed."
+  nullable    = false
+}
+
+variable "primary_shared_key" {
+  type        = string
+  description = "Primary shared key of the central monitoring LAW."
+  sensitive   = true
+}
```

Comment:
Why does it need to be the **primary** shared key?

And why not just expect it as part of the `log_analytics_workspace` object?

Reply:
@QBY-MarkusMaring is this cleaner than having it in the LAW variable?

Reply:
The primary_shared_key is not part of the CAF output. This input will likely be set by the CAF law_output.

Also the extra variable allows us to only mark that one as sensitive. It doesn't need to be the primary one tho. "law_shared_key" would make more sense.